### PR TITLE
Fix closures

### PR DESCRIPTION
**Pull request by orangeturtle739**
Code like https://github.com/cornell-brg/pymtl/compare/jng55-closures?expand=1#diff-ae2a0fffb8117a39395422133340c31fL29 now works!

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -43,7 +43,7 @@ install:
  # Install Python requirements
 
  - pip install --editable .
- - pip install --upgrade pytest
+ - pip install --upgrade 'pytest<=3.9.1'
```
Comment (orangeturtle739) [inline .travis.yml]: See https://travis-ci.org/cornell-brg/pymtl/jobs/456738729#L574.

**Comment by cbatten** [comment]
@jsn1993 please take a look and see what you think.

**Comment by cbatten** [comment]
tell me more about the pytest upgrade. does that mean that anyone that is using pymtl on travis will no need to change their travis.yml file? (i.e., ~100 4750 students?)

**Comment by orangeturtle739** [comment]
I don't think everyone who is using pymtl on travis will have to change `.travis.yml`, but I'm not positive. It seems that the tests for pymtl itself are using some deprecated feature, which was removed in a new version of pytest. But the students in 4750 shouldn't be running the pymtl tests themselves, so they should be OK.

File: pymtl/tools/ast_helpers.py [hunk 2]
```diff
@@ -112,3 +112,15 @@ def get_method_ast( func ):
...
   closure_objects = [c.cell_contents for c in fn.func_closure]
   return dict( zip( fn.func_code.co_freevars, closure_objects ))
+
+#-----------------------------------------------------------------------
+# get_default_arg_values
+#-----------------------------------------------------------------------
+# returns a dictionary of arg_name:default_values for the input function
+# https://stackoverflow.com/a/12627118
+def get_default_arg_values(func):
+    args, varargs, keywords, defaults = inspect.getargspec(func)
```
Comment (jsn1993) [inline pymtl/tools/ast_helpers.py]: BRG Python guideline suggest we write func( x, y ) with spaces around the arguments. Please take a pass through the codebase.

File: pymtl/tools/ast_helpers.py [hunk 3]
```diff
@@ -112,3 +112,15 @@ def get_method_ast( func ):
...
+    if defaults is None:
+      defaults = []
+    return dict(zip(reversed(args), reversed(defaults)))
```
Comment (jsn1993) [inline pymtl/tools/ast_helpers.py]: same here

File: pymtl/tools/ast_helpers.py (hunk 2 above)
Comment (jsn1993) [reply pymtl/tools/ast_helpers.py]: Sorry, I mean whatever you added/modified.

**Comment by cbatten** [comment]
so what do we need to do next to merge this in?

**Comment by jsn1993** [comment]
> so what do we need to do next to merge this in?

I'm just asking for spacing fixes. After anyone push a commit to fix all spacings, I will approve.

**Comment by cbatten** [comment]
right ... i posted a comment about that. for this specific pull request since it is just an update to pymt v2 maybe it is fine as it is?

**Comment by orangeturtle739** [comment]
@cbatten @jsn1993 fixed the spacing

**Comment by jsn1993** [comment]
LGTM

**Pull request merged**